pkg/prometheus/server/thanos_sidecar_config.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheus
+
+import (
+	"fmt"
+
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	prompkg "github.com/prometheus-operator/prometheus-operator/pkg/prometheus"
+)
+
+const (
+	thanosConfigDir                                  = "/etc/thanos/config"
+	thanosPrometheusHTTPClientConfigFileName         = "prometheus.http-client-file.yaml"
+	thanosPrometheusHTTPClientConfigSecretNameSuffix = "thanos-prometheus-http-client-file"
+)
+
+type thanosPrometheusHTTPClientConfig struct {
+	TLSConfig  *monitoringv1.TLSConfig
+	secretName string
+}
+
+func newPrometheusHTTPClientConfig(p *monitoringv1.Prometheus) *thanosPrometheusHTTPClientConfig {
+	return &thanosPrometheusHTTPClientConfig{
+		TLSConfig: &monitoringv1.TLSConfig{
+			SafeTLSConfig: monitoringv1.SafeTLSConfig{
+				// sidecar is listen to prometheus server on localhost in pod, no need to use tls.
+				InsecureSkipVerify: ptr.To(true),
+			},
+		},
+		secretName: thanosPrometheusHTTPClientConfigSecretName(p),
+	}
+}
+
+// CreateOrUpdatePrometheusHTTPClientConfigSecret create or update a kubernetes secret with the data for thanos sidecar
+// communicated with prometheus server.
+// https://thanos.io/tip/components/sidecar.md/#prometheus-http-client
+func (c *thanosPrometheusHTTPClientConfig) createOrUpdatePrometheusHTTPClientConfigSecret() (secret *v1.Secret, err error) {
+	dataYaml := yaml.MapSlice{}
+	dataYaml = append(dataYaml, yaml.MapItem{
+		Key: "tls_config",
+		Value: yaml.MapSlice{
+			{
+				Key:   "insecure_skip_verify",
+				Value: c.TLSConfig.InsecureSkipVerify,
+			},
+		},
+	})
```

Comment:
(nit) I realize that we don't really need a constructor and a simple function would be more compact.
```suggestion
// buildPrometheusHTTPClientConfigSecret returns a Kubernetes secret with the HTTP configuration for the Thanos sidecar to
// communicate with the Prometheus server.
// https://thanos.io/tip/components/sidecar.md/#prometheus-http-client
func buildPrometheusHTTPClientConfigSecret(p *monitoringv1.Prometheus) (*v1.Secret, error) {
	dataYaml := yaml.MapSlice{}
	dataYaml = append(dataYaml, yaml.MapItem{
		Key: "tls_config",
		Value: yaml.MapSlice{
			{
				Key:   "insecure_skip_verify",
				Value: true,
			},
		},
	})
```

Reply:
yeah, it will make the func more clear and easier to read.